smack-extensions/src/main/java/org/jivesoftware/smackx/pubsub/PubSubManager.java

```diff
@@ -307,7 +307,8 @@ private LeafNode getOrCreateLeafNodeProsodyWorkaround(final String id)
             return createNode(id);
         }
         catch (XMPPErrorException e1) {
-            if (e1.getXMPPError().getCondition() == Condition.conflict) {
+            if (e1.getXMPPError().getCondition() == Condition.conflict ||
+                    e1.getXMPPError().getCondition() == Condition.service_unavailable) {
```

Comment:
Do we need the service-service unavailable check here? I'd expect prosody and most servers to either create the node or throw a conflict error.